server/service/src/programs/patient/README.md

```diff
@@ -0,0 +1,69 @@
+
+## Legacy mSupply patients
+
+This applies where:
+
+- Patient was created on Legacy mSupply site, hasn't been synced to any OMS sites yet
+- Patient is on an OMS remote site, but that site hasn't yet upgraded to 2.7 (so as far as Central OMS is aware, patient doesn't exist, and only exists on Legacy mSupply)
+
+Patient fetch calls the [link_patient_to_store mutation](/server/graphql/programs/src/queries/link_patient_to_store.rs), which makes a call to Legacy mSupply to add name_store_join between the patient and the calling store. This queues all the related patient records for the next time the remote site syncs.
+
+It then calls OMS Central `name_store_join`, which adds the same `name_store_join` to OMS Central (so OMS Central can also keep track of patient visibilty.) However, `name_store_join` requires name/name_link to exist on OMS Central, and we realise at this point that it does not yet.
+
+- OMS Central then makes its own call to Legacy name_store_join endpoint, to add patient visibility for itself (a store on central server)
+- It then triggers a sync cycle (OMS Central to Legacy) to receive the name record for this patient
+- Only after this is the name_store_join record for the remote store inserted, and success response to the remote site
+
+Risks: the `link_patient_to_store` is a one time mutation, and can't be done in a transaction. If there are connection issues, we might end up with name_store_join added on Legacy, but not OMS Central. This would not be apparent to the end user, would prevent v6 patient records from syncing between sites. Would be hard to diagnose... TODO?
+
+- Both central servers are on same server, if can connect to one, should be able to connect to both, and shouldn't be a connectivity risk between the two central servers. This can't be guaranteed in all cases though.
+
+## Re-initialising a site
+
+After above migrations, patient `name` and `name_link` records should exist on OMS Central server, vaccination changelogs should have correct `name_link_id` set, and `name_store_join` record for remote store to patient should exist. When site re-initialises, vaccination records should be included via name_store_join.
+
+There is a small use case - remote site is upgraded 2.7, but re-initialised before syncing. If there are vaccination records, they will be orphaned. (Per point 3 of 2.7 Changes, on OMS central the vaccination changelogs wouldn't have name_link_id, so wouldn't get re-fetched on initialisation.) Given low likelihood of existing vaccination records, and of reinitialising before a sync, we can ignore this case.
+
+## New patient created in OMS remote site
+
+Creating a new patient on OMS remote site will create new name and name_store_join records. We sync this to both Legacy mSupply Central and OMS Central. All future records will have constraints met.
+
+We should probably ensure OMS Central adds its own name_store_join for the new patient, for if patient is merged etc.
```

Comment:
I'm wondering if name translation should be extended somehow - if we have a translator mode, basically "I am running as OMS central, integrating records from OMS remote site" - at this point name translation could do OG `/nsj` add, to add visibility for patient to central store

Reply:
Created: https://github.com/msupply-foundation/open-msupply/issues/7347